### ISSUE TITLE
New Skin contribution enquiry

### ISSUE THREAD

**Issue by coddec** [Open-Shell/Open-Shell-Menu#538]
By @bonzibudd, posted on Discord in general channel

>Apologies if this isn't a good place to put this, but I figured it was better than nothing.
>
>I am currently working on a new skin, which I think would be a very good addition for Open-Shell on Windows 10. If the maintainers are interested, I think this would be a good upgrade from the old Metro skin as the default. These will be the main improvements:
>
>-Updated iconography. Current icons/arrows are an inconsistent mix of Win7 to Win8, and some custom.
-Ability to change main menu color using "Override" in OSM Settings. This is a downside of the current Metro theme.
-Modern color schemes. The new skin will use the Windows mode to determine some aspects of the menu, including sub-menus and search boxes. I might add this to the main menu as well.
-Improved scalability of icons and paddings. In fact, many items in Metro theme don't scale at all.
-Consistency between elements of opaque and transparent modes (Separators, selectors, etc.)
-Better implementation of Large font option
-More modern fonts (ex. SegUI semibold) in some areas
-More options, some ported from other skins (black text, white submenu, etc.)
>
>This would not involve removing the Metro theme entirely, but just changing the default. Obviously, the maintainers could review it just to make sure there aren't any issues.
>
>Thank you for your consideration.

@ge0rdi @Ibuprophen @XenHat @coddec

Thanks a lot for you interest and contribution BTW 😊

**Comment by bonzibudd** [comment]
Thanks for this! I'll continue to give updates on the status here. 

This will be for the classic and 7-styles, so switching between the interfaces will be seamless for users.

What I've made so far is basically a more "unified" metro theme, I've updated the selectors and some colors, and adjusted bitmaps accordingly to look a bit nicer. 

Something else about the Metro theme, it tends to have some contrast issues on some colors, due to the mix of some colors that sometimes overlap. I'll probably have to change those out.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/99859629-944e1f00-2b5e-11eb-87d8-ffdd08aa5362.png)

Improvements so far. Background color now respects the "Override glass color" setting to improve customization. Next will probably be adding the "Immersive" colors in the search box and submenus.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/99885944-bb602b80-2c06-11eb-9afc-4e6e588976d2.png)

Added the immersive colors to submenu, now supports auto light-dark. Bitmap updated to accommodate.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/99921544-09555c00-2cf9-11eb-9757-432ea6d9df82.png)

Win7 style so far. I'll need to fix up the search bar quite a bit, but the auto colorization does work now. I'll probably give it a border, either accent color or gray, and the magnifier icon color will probably be changed.

Overall, still looks kinda bland, but it is coming along. I'll be working on adding some more Win10 glyphs next.

**Comment by bonzibudd** [comment]
Here's what the 7 style currently looks like:
![image](https://user-images.githubusercontent.com/61938331/100043096-b3f17b80-2dda-11eb-9a0d-85c627180e6a.png)

So far, I've done all of this:

 - Updated all existing icons controlled by the skin. This includes the arrow, magnifier, pin, all programs, and pager icons, which are now the same versions used in Windows 10. 
 - Added automatic dark/light to submenu and search box
 - Gave high DPI options to more icons
 - Switch from black background bitmap to #FF0000, allowing for custom color overrides to be set
 - Made all aspects of menu scalable, excluding those which rely on a bitmap (separators, user image)
 
I still have a few things I would like to do before this could be added natively:
 - More options, including multiple transparency modes, custom font sizes, reduce glass color, black text on glass, and white sub-menus (possibly)
 - Bitmaps for 120DPI
 - More tweaks for "Small icons" option

While I'm here, I'll go ahead and upload the files:
[Win10 Skin.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5587339/Win10.Skin.zip)

They should be fully functional as of now.

**Comment by ge0rdi** [comment]
Few personal comments:
- I find that blue border of search box quite distracting. Can't it have the same color as other separators?
- "Back" icon is blue too (was that intended?)
- Scrollbar is too bright imo
- Also the "Shutdown" button looks too bright (comparing to the rest of menu)

I'm using `Tenified` skin (can be easily googled) that fits Win10 really nicely.

**Comment by bonzibudd** [comment]
@ge0rdi 

1. I can definitely change the color of the search box border to be a more similar gray. I'm still experimenting with some of the colors, so some items are subject to change.
2. Yep, I've also fixed that. It's a remnant of when I changed the programs bitmap and mask colors.
3. Yeah, the scrollbar looks pretty bad right now, it currently uses a Windows 8 style color scheme from the Metro theme. In theory this should match the native Start menu.
4. Should be an easy fix.

I appreciate the input.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100124083-3153d500-2e49-11eb-838b-0662bd093894.png)
Couple tweaks, search box border is now separator color and shutdown button is less bright. There might need to be an in-between color for the shutdown box to make sure it stays visible on lighter colors. I may also choose to remove the border completely.

**Comment by CTVCAM8** [comment]
Possible issue with some people.
The closeness of the magnifier and "X" is a bit scary.
It's ok to have close in 2 columns style as the submenu covers up the shutdown selector.

On 7 style, it's too easy to for sloppy mouse actions to miss the X and hit the shutdown button!

**Comment by bonzibudd** [comment]
@CTVCAM8 good call. That is definitely one scary shutdown button...

I'm actually making all of the separators taller/wider overall, so this will go well with that change. If I need to, I will add some of that horizontal search padding back.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100180659-9c7fc480-2ea6-11eb-89bc-cbe398197d4a.png)

![image](https://user-images.githubusercontent.com/61938331/100180452-2f6c2f00-2ea6-11eb-9d0b-e7e02b328f03.png)

I changed the overall width slightly, and that padding now scales better. It will still require more spacing, but I'd like to keep it in line with the other vertical separators for now.

**Comment by CTVCAM8** [comment]
Another way is to push the shutdown button away with a few pixels using shutdown padding.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100298298-c5fc2700-2f5e-11eb-803b-03d6b8568003.png)
Scrollbar background will look about like this. Currently matching opacities and colors to match Win10's native scrollbar.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100307912-05357280-2f75-11eb-95db-7caac35fad34.png)
Scrollbar is improved. Should fit in well with the rest of the OS, however transparency seems wonky with glass disabled. I might have to look into that later.
![image](https://user-images.githubusercontent.com/61938331/100308755-24cd9a80-2f77-11eb-8300-abca0557e4ff.png)

**Comment by bonzibudd** [comment]
Here's another updated version, I've changed some paddings and introduced better support for 120DPI. 

[OSM skins.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5604636/OSM.skins.zip)
![image](https://user-images.githubusercontent.com/61938331/100385467-aca3bb00-2ff0-11eb-828e-7988c4fd777d.png)

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100406849-edbdbe80-3034-11eb-9db0-df1dbefd06db.png)
I added a new "Reduce glass color" option, it should behave similar to the same option in other skins.

This gives the menu a Vista-like color bleed, and users of the Win7/8 skins will be familiar with this.

Next, I am planning on adding an option for dark text/icons/selections in the main menu using the labels of existing string resources. This will effectively allow users to choose a light theme in the main menu.

Another option I am considering is a two-tone style layout, in which the left-hand column follows the dark/light theme. However, this may require some padding changes to look good, so I will hold off on this for now.

**Comment by bonzibudd** [comment]
Here's that light theme I was talking about:
![image](https://user-images.githubusercontent.com/61938331/100461953-8127db80-3097-11eb-9cad-bfe747a47d49.png)

**Comment by bonzibudd** [comment]
Classic style for reference:
![image](https://user-images.githubusercontent.com/61938331/100463822-74f14d80-309a-11eb-898d-db79d2db9d6f.png)
![image](https://user-images.githubusercontent.com/61938331/100463919-9a7e5700-309a-11eb-9bcf-c9f26fb142bb.png)
This allows for more flexibility for customizing the color scheme.

**Comment by bonzibudd** [comment]
[OSM Skins more options.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5609187/OSM.Skins.more.options.zip)

This version has more options, including black text/selectors, custom font size, and reduce glass color. I also made some changes for higher DPIs.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/100521113-a851dc00-316f-11eb-8960-c7f91ac58bfc.png)
Slight change: Main and submenu arrows are now a different color until hovered over or selected. This is more similar to how Win10's menus behave, and it looks a bit less jarring. I also did something similar to the search bitmap on the classic style, as I thought it stood out too much.

**Comment by bonzibudd** [comment]
[OSM Skins arrow mask.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5633329/OSM.Skins.arrow.mask.zip)

Not too much different in this version, just a couple fixes for a setting and improvements to the arrows as previously mentioned.

I still have a few things I want to change (paddings and options), but I think it's in a good spot right now. If the maintainers are actually interested in adding this at some point, I would be interested to hear some feedback.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/101701157-5735c800-3a4c-11eb-8a2f-996dd87686d9.png)
I've begun work on the "Two-tone" option I previously mentioned. For reference, the part that is currently black will automatically change with light/dark, similar to the submenu.

Can't say if I will end up keeping this, but it's definitely doable. However, the classic style will be more work, since it will involve a custom background when `User_image` and `User_name` are in use. Speaking of User_name, I might try to get that to work on the Win7 style as well.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/101990312-4798c800-3c74-11eb-994b-08671f5eb786.png)
New "User name" and "Center user name" options for the 7-style.

Something I noticed is that you can't override the user name while in the Windows 7 mode, you have to change it in the Classic styles and come back again. This is probably because the user name isn't "officially" supported for the 7-styles.

**Comment by bonzibudd** [comment]
Demonstration of light/dark modes: 
![image](https://user-images.githubusercontent.com/61938331/102235255-24625880-3ec0-11eb-8956-17a8d0343fdc.png)
![image](https://user-images.githubusercontent.com/61938331/102235354-4065fa00-3ec0-11eb-9bb2-753dba60463b.png)
You can't override the background color of left-hand side in this mode, but the right-hand side acts the same as before. You can still switch back to the fullglass "Metro" style as well.
![image](https://user-images.githubusercontent.com/61938331/102235774-b0748000-3ec0-11eb-8a47-539a8accb357.png)

**Comment by bonzibudd** [comment]
I've been trying to decide on a sleeker outline for the two-tone style. For now, I've added a thin border which is a similar color to the separator, which is in a more similar design to Win10.
![image](https://user-images.githubusercontent.com/61938331/102640401-67266980-4128-11eb-9ccf-c87bdb363049.png)

Just to see what would happen, I tried the skin on High Contrast mode:
![image](https://user-images.githubusercontent.com/61938331/102641879-d00ee100-412a-11eb-8afa-0a722ab7a373.png)

Not terrible actually, and even this will be improved with time as I add better selectors and button masks.

Other changes: Text padding/height/width. The goal is to have more Win10-like padding without sacrificing how many items can fit in a similar area. Submenus and lists will remain the same.

**Comment by bonzibudd** [comment]
Some thoughts about the classic w/ two columns style:
![image](https://user-images.githubusercontent.com/61938331/102716418-da0b1e00-42a9-11eb-8e2e-7bdcfcc5bd56.png)

 - I may end up changing the top user bar so that it only aligns to the right side, just like the Metro theme. The reason that I changed this in the first place was to make a better distinction between the classic and 7-styles, and appear more similar to Windows XP. While I personally prefer this style, it has a limitation that I will describe here:
 - It will be much more difficult to add the two-tone style to the main menu, since I will need a new bitmap when user image and user name are enabled. There will also need to be a new set of bitmaps for different DPIs.

For now, I will make a copy of that skin and see if I can make do with the current layout. If not, I will go back the right-aligned layout.

**Comment by bonzibudd** [comment]
Quick look at classic style (experimental style):
![image](https://user-images.githubusercontent.com/61938331/102724644-f5445080-42de-11eb-844f-c0dc872b6864.png)

**Comment by bonzibudd** [comment]
Slightly updated classic style:
![image](https://user-images.githubusercontent.com/61938331/103154781-f5719f80-4767-11eb-918c-69a9944d92a2.png)

I'll do something else with the user bar at the top once I get all of the new bitmaps aligned.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/103160309-3f2aac00-47a1-11eb-8d32-0cef9b873c28.png)

New style more finalized, now works on all DPIs and all other configurations in theory. 

The immersive background still needs to be added to the Win2000 style, but that will wait until I find a good solution for the user bar at the top. 

After that, the next change will probably be the selectors and pagers on the immersive background, they will be similar to the submenu.

There are some more bitmap and layout changes that need to be made, but this is getting close to the "final" version.

**Comment by Darthagnon** [comment]
These look awesome! Are these changes all part of your [Fluent Metro](https://github.com/bonzibudd/Fluent-Metro) skin?

**Comment by bonzibudd** [comment]
@Darthagnon 

I actually just came back to Fluent-Metro today after taking a break to work on this. These skins are very different, but I have been working on bringing the colorization and bitmap improvements over. Also many changes that may not be very obvious, but will hopefully make it easier to add more features in the near future.

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/103952757-eea33f00-510e-11eb-93b3-ef5d6c920c74.png)

Back from a small break. I have been trying to find a decent solution for the user "bar" for the classic styles. For now I have settled on a transparent bar which uses the immersive light/dark colors. I kept a similar border around the outside to match better as well. This may still change in the future, but for now I will add these colors to all of the necessary bitmaps. The next step will be to add this new style and colors to the one-column style.

After that, I will be working on new selectors/frames/pagers for the left-hand side to match Windows 10, similar to how I did the submenus. This may take a while just because there is not a definitive way to calculate these colors/ratios, but it will be a needed improvement. Glass selectors will stay the same.

I don't have anything else planned for this skin, so I will consider this finished when I complete the above. I obviously don't really have a say as to if this skin is added into Open-Shell natively, but I think that these improvements will help give Open-Shell a breath of fresh air for old and new users.

**Comment by ricardoteixeiraduarte** [comment]
(This skin is great!)
Any ideia why I'm getting black text on the left side? 
It's unreadable.

![image](https://user-images.githubusercontent.com/7994937/107064500-798d5d00-67d3-11eb-971f-c8194074ca44.png)

**Comment by bonzibudd** [comment]
@ricardoteixeiraduarte I have some ideas.

What version of Windows 10 are you running? I think there could be an issue with how older versions handle the immersive colors.

Also, post a screenshot of your "Skins" tab if you can.

**Comment by ricardoteixeiraduarte** [comment]
It's Windows Server 2012. Maybe that's the issue.

![image](https://user-images.githubusercontent.com/7994937/107066299-b1959f80-67d5-11eb-89a6-86dabbd3a1ff.png)

**Comment by bonzibudd** [comment]
Ok, I see. Sadly I don't think I will be able to make this truly compatible on older versions of Windows due to the lack of some system colors which only work with Windows 10. 

You could try changing the Transparency level option to "More transparent," but I can't guarantee it will fix it on all surfaces. I'll consider making a version with static colors in the future.

**Comment by ricardoteixeiraduarte** [comment]
"More Transparent" does make the text color white, but loses the "two tone".
Looks like this:
![image](https://user-images.githubusercontent.com/7994937/107162498-7a70eb00-699b-11eb-9b81-8b0c6a26849d.png)

**Comment by ricardoteixeiraduarte** [comment]
Are you able to implement a setting like "White text on non-glass"?
That would fix the issue.

**Comment by bonzibudd** [comment]
Yes, I could add something like that, however that would in practice break on Windows 10 when the system background color changes. It would have to detect the version of Windows to apply those settings, which isn't currently possible in Open-Shell. Everything needs to work automatically.

Don't worry though, because I will make a different version for Windows 7/8 using the old color system. That will be priority when I finish the current version.

**Comment by bonzibudd** [comment]
[More improvements.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5974242/More.improvements.zip)

Alright, here is a more final look at the skins. I made some padding and selector changes to be closer to Windows 10, along with some other general changes. I have also began adding the immersive main background to the one-column menu, but that version is currently unusable because of the background bitmaps.

If anyone has any suggestions or problems with the skins, please point them out. The one-column style only works in "More transparent" mode, so also keep that in mind.

**Comment by bonzibudd** [comment]
Also, I'm going to be installing some older versions of Windows 10 to test the immersive colors. If there are any issues, there may need to be some checks for the specific build of Windows.

**Comment by bonzibudd** [comment]
Good news, the colorization works fine back to version 1507. I will also try 1903 to see if the light theme works. 1507 to 1809 will be forced to use the dark colors, which should be fine as it matches the taskbar.

![image](https://user-images.githubusercontent.com/61938331/107834221-a2799900-6d63-11eb-9c98-5d8b8c0838a6.png)

**Comment by bonzibudd** [comment]
![image](https://user-images.githubusercontent.com/61938331/107837062-3ef46900-6d6d-11eb-9f7e-acf1db546f30.png)

1903 also seems to work fine.

**Comment by bonzibudd** [comment]
Hey @ricardoteixeiraduarte, I just had a realization. I think it will, in fact, be possible to make compatibility work on Windows 8. Open-Shell allows for use of "Compatible" metro colors for switching between different OS's, which I completely forgot about. I will begin work on that tomorrow.

**Comment by bonzibudd** [comment]
I was, in fact, able to get Win8 compatibility improved, so that helps quite a bit. It will be forced to use light colors, but that should still be an improvement for the time being.

Additionally, I have been working on immersive colors for the one-column style, and you can see that here:

![image](https://user-images.githubusercontent.com/61938331/107858841-4ca8fd80-6e04-11eb-9bcc-869ceff40367.png)

**Comment by ricardoteixeiraduarte** [comment]
Great news!
I will give it a try as soon as you make it available.

**Comment by bonzibudd** [comment]
[One column immersive.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/5978435/One.column.immersive.zip)

Here's a somewhat more finalized look at the skin, this fixes the one-column styles with the default background, and has some minor padding changes. At this point, I know of a few formatting issues in the skin file itself, but I haven't found any usability bugs yet. Overall, there could still be a range of just a few or a lot of changes to make, so all feedback is welcome.

@XenHat 
@ge0rdi 
@coddec 

If you all are interested, I would appreciate any suggestions or bugs you come across, and I would also be curious if there is any interest in adding this to Open-Shell by default once it's ready. I don't know if I would need to submit a pull request or something else, but I'm happy to contribute any way that I need to.

**Comment by ricardoteixeiraduarte** [comment]
It is now working fine with Windows Server 2012.

One small issue though. On the following screenshot it looks like the non-glass area is one pixel short of vertical size:
![image](https://user-images.githubusercontent.com/7994937/107889333-a7784d00-6f09-11eb-93b1-babcf6651d43.png)
Are you able to remove the top border from the non-glass area? Or make it the same color as the rest of the non-glass area?

Another question: how is the color for the glass area defined? Can it be changed on the settings?

Thanks.

**Comment by ricardoteixeiraduarte** [comment]
It seems that by setting "reduce glass color", the glass color no longer follows the windows "theme".
Disabling "reduce glass color" makes the menu follow the colors, but it becomes too light for white text.

**Comment by bonzibudd** [comment]
@ricardoteixeiraduarte 

The border is definitely something I can change, thanks for pointing that out. On that note, I may make some changes to the color of the user name background, just to make it more visible from the rest of the menu.

The glass color can be changed by the "Override glass color" option under the Menu Look tab.

**Comment by ricardoteixeiraduarte** [comment]
Another thing I saw is that if I select "More Transparent" without "Reduce Glass Color", the program list will not follow the menu color but stays black.

**Comment by bonzibudd** [comment]
I have seen that issue before, and it seems to be another Windows 8/2012 specific issue. Try setting the "Menu color blending" option to 0 and see if that makes a difference. 
![image](https://user-images.githubusercontent.com/61938331/107890542-8dc60e00-6ee7-11eb-9b85-d840efb2d8bc.png)

The final version might not end up supporting Windows 8/2012 under Open-Shell because of these issues, but I will address as many as possible. You will still be able to use versions uploaded here, however.

**Comment by ricardoteixeiraduarte** [comment]
It is working fine.
About that "top border issue", one option would be to use the same border color for all the start menu, instead of different border color for glass and non glass.
That way, that "1px" effect would no longer be present.

**Comment by bonzibudd** [comment]
No problem, I will make a few changes and see what works the best for all configurations.

**Comment by ricardoteixeiraduarte** [comment]
Other than that small issue, everything seems perfect. 
Amazing work!

**Comment by bonzibudd** [comment]
If anyone's wondering, I am still working on this skin. However, I'm probably going to be making some changes to my approach to how some aspects of it will work out. The overall plan is still similar, I just want to make some improvements to better future-proof it.

Some things I have in mind right now:

 - Scale icons at 200%
 - Use the new Fluent icon set
 - Make backwards compatibility a bigger focus (Win8 specifically)
 - More thorough customization (background colors, etc.)

**Comment by bonzibudd** [comment]
Forgot to add: I'll probably be switching the resources to PNG as well for a smaller file size.

**Comment by XenHat** [comment]
How's it going?

**Comment by bonzibudd** [comment]
@XenHat It is coming along well so far, I'm not at my computer right now but I'll try to post more information later. There are still many small things I need to experiment with to try to get it as close to "native" looking as possible.

**Comment by bonzibudd** [comment]
I apologize for not keeping up with this; I've been very busy in my personal life recently and I haven't had much time to do what I want with this and other things regarding Open-Shell. I don't know if I will have the time to finish this in the near future, as there are still many changes I'd like to make before finalizing it. However, feel free to keep using the files I've provided as-is, just don't expect any changes for a little while.

**Comment by bonzibudd** [comment]
For anyone interested in this skin's development, I have an updated version:
[PNG skins.zip](https://github.com/Open-Shell/Open-Shell-Menu/files/10328319/PNG.skins.zip)

Here are the changes:
 - Updated option strings to reflect changes better
 - Added ability to override system color
 - Added 200% scaling icons
 - Fixed texture for submenu split selectors
 - Fixed texture for system colored pagers in main menu
 - Caption now scales indefinitely
 - User name scales up to 250%
 - Images use PNG for decreased file size

**Comment by CTVCAM8** [comment]
I forgot about your 2-tone skins.

A lot of people would never know just how many classic-style menu images 
are required for the caption, user name, and user picture for all the sizes involved. 

The skins look ok with a quick lookover.

The scrollbar arrow is a bit dark with black buttons and full glass options though.

![Scroll bar arrow](https://user-images.githubusercontent.com/44127216/210181994-03241586-5676-486b-af6b-f04b516ad951.png)

**Comment by bonzibudd** [comment]
@CTVCAM8 

Hm, I'm not getting that on my system. The arrow background should be the same color as the rest of the scrollbar background. I'll have to look into that.

**Comment by bonzibudd** [comment]
Nevermind, I can see the issue now. I will see if I can improve the appearance.

**Comment by CTVCAM8** [comment]
I notice at line: 206 there is '#' instead of '$' used in front of a Metro color.
I don't think it will matter in this case.

Good that you found the problem.

Also, check out arrow brightness goes the opposite in this case as well.
If just the black text option is enabled, the arrow gets brighter.

![Only black text option](https://user-images.githubusercontent.com/44127216/210183722-1f7ae0b0-223d-44f6-a973-608c6631c0ba.png)


If black text, black buttons options enabled, the arrow gets brighter. 
So that works ok.

![Blacktext and black buttons](https://user-images.githubusercontent.com/44127216/210183727-5260dd6c-94eb-4059-a71c-21136f2f8009.png)

**Comment by bonzibudd** [comment]
OK, I fixed it, it should look like this now:
![image](https://user-images.githubusercontent.com/61938331/210183712-f922b1ef-8245-475e-8ac5-0cfa927cd2e4.png)

I did also see the issue with black text without using black buttons, it is fixed as well. The color normally would invert at the 3rd state, but in the case of these two options it messed up the visibility.

I also noticed that mistake in the `Scrollbar_arrows` line, but that is actually a leftover from the original Metro theme. It shouldn't matter because the arrows are always using a tint color instead. Maybe Ivo did that intentionally as a placeholder.
